pkg/apiserver/request/requestinfo.go

```diff
@@ -103,7 +103,7 @@ type RequestInfoResolver struct {
 // /api
 // /healthz
 // /
-func (r *RequestInfoResolver) GetRequestInfo(req *http.Request) (RequestInfo, error) {
+func (r *RequestInfoResolver) GetRequestInfo(req *http.Request) (*RequestInfo, error) {
```

Comment:
it makes it very clear that this is not a shared instance. the caller is free to take the address of the returned struct if it wants, but I'd rather return a clearly unshared struct.